homeassistant/components/teslemetry/button.py

```diff
@@ -0,0 +1,114 @@
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .entity import TeslemetryVehicleEntity
+from .models import TeslemetryVehicleData
+
+
+@dataclass(frozen=True, kw_only=True)
+class TeslemetryButtonEntityDescription(ButtonEntityDescription):
+    """Describes a Teslemetry Button entity."""
+
+    func: Callable | None = None
+
+
+DESCRIPTIONS: tuple[TeslemetryButtonEntityDescription, ...] = (
+    TeslemetryButtonEntityDescription(key="wake"),  # Every button runs wakeup
+    TeslemetryButtonEntityDescription(
+        key="flash_lights", func=lambda self: self.api.flash_lights()
+    ),
+    TeslemetryButtonEntityDescription(
+        key="honk", func=lambda self: self.api.honk_horn()
+    ),
+    TeslemetryButtonEntityDescription(
+        key="enable_keyless_driving", func=lambda self: self.api.remote_start_drive()
+    ),
+    TeslemetryButtonEntityDescription(
+        key="boombox", func=lambda self: self.api.remote_boombox(0)
+    ),
+    TeslemetryButtonEntityDescription(
+        key="homelink",
+        func=lambda self: self.api.trigger_homelink(
+            lat=self.coordinator.data["drive_state_latitude"],
+            lon=self.coordinator.data["drive_state_longitude"],
+        ),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Teslemetry Button platform from a config entry."""
+
+    async_add_entities(
+        chain(
+            (
+                TeslemetryButtonEntity(vehicle, description)
+                for vehicle in entry.runtime_data.vehicles
+                for description in DESCRIPTIONS
+                if Scope.VEHICLE_CMDS in entry.runtime_data.scopes
+            ),
+            (
+                TeslemetryRefreshButtonEntity(vehicle)
+                for vehicle in entry.runtime_data.vehicles
+            ),
+        )
+    )
+
+
+class TeslemetryButtonEntity(TeslemetryVehicleEntity, ButtonEntity):
+    """Base class for Teslemetry buttons."""
+
+    entity_description: TeslemetryButtonEntityDescription
+
+    def __init__(
+        self,
+        data: TeslemetryVehicleData,
+        description: TeslemetryButtonEntityDescription,
+    ) -> None:
+        """Initialize the button."""
+        self.entity_description = description
+        super().__init__(data, description.key)
+
+    def _async_update_attrs(self) -> None:
+        """Update the attributes of the entity."""
+
+    async def async_press(self) -> None:
+        """Press the button."""
+        await self.wake_up_if_asleep()
+        if self.entity_description.func:
+            await self.handle_command(self.entity_description.func(self))
+
+
+class TeslemetryRefreshButtonEntity(TeslemetryVehicleEntity, ButtonEntity):
```

Comment:
Why is this one separate?

Reply:
Oh, I see now. Hmmmmm, We usually want people to use the `update_entity` service for this. (but that doesn't explicitly wake the vehicle probably)

Reply:
Yeah, I had this requested because its a very common thing people want to do, if its "against the rules" ill drop it and tell people to use a script to call wake_up when required before update_entity.